src/GenesisMakeFileWindow.cpp

```diff
@@ -137,21 +158,67 @@ void GenesisMakeDirWindow::MessageReceived(BMessage* message)
 }
 
 ////////////////////////////////////////////////////////////////////////
-bool GenesisMakeDirWindow::CreateFolder(const char *dirpath, const char *dirname)
+bool GenesisMakeFileWindow::CreateFolder(const char *dirpath, const char *dirname)
 ////////////////////////////////////////////////////////////////////////
 {
 	BString dir;
 	dir.SetTo(dirpath);
 	dir+="/";
 	dir+=dirname;
 
-	if (create_directory(dir.String(), 0777)==B_OK)		// TODO: jo a 0777?
+	BEntry *entry = new BEntry(dir);
+	if (entry->Exists() && entry->IsDirectory())
```

Comment:
You may want to test, if the existing entry is a symlink to a folder. When the symlink is a file, it works: the Entry-already-exists-Alert pops up. In case of symlinked folder, nothing is shown.